### PR TITLE
[PF-541] refactor kube pod listener

### PR DESCRIPTION
**Pull request by jaycarlton**
I was studying this class and had a little trouble following what it does, so I broke it into some  smaller functions. Also added `final`, since some variables are imperative and some  aren't, some stream and Optional usage, and a couple of comments.

### CODE REVIEW

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java [hunk 1]
```diff
@@ -111,34 +111,16 @@ public void run() {
...
-                  }
-                } catch (DatabaseOperationException | StairwayExecutionException ex) {
-                  logger.error("Stairway recoverStairway failed to recover pod: " + podName, ex);
-                } catch (InterruptedException ex) {
-                  logger.info("KubePodListener interrupted - exiting", ex);
-                  exception = ex;
-                  return;
-                }
-              }
+            if (handlePodOperation(operation, podName)) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: This is the most awkward bit. It returns a boolean letting you know to return early, but the name almost needs to be `handlePodOperation_isInterrupted`. 🤢

**Comment by jaycarlton** [comment]
Looks like there's no unit test coverage for this guy, so I'm thinking I should add some. The mocking might be wonky but I think it should work.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java [hunk 2]
```diff
@@ -86,22 +88,20 @@ public void run() {
     exception = null;
     int consecutiveRetryCount = 0;
     int retryWait = WATCH_INITIAL_WAIT;
+
+    // retry loop
     while (true) {
       try {
-        ApiClient client = Config.defaultClient();
+        final ApiClient defaultClient = Config.defaultClient();
```
Comment (ddietterich) [inline src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: I prefer client. There is no other client, so calling it default doesn't inform anything.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: How about
- make a void method for Add
- make a void method for Delete
- move the InterruptedException catch to the outer try and return from there.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java [hunk 3]
```diff
@@ -111,34 +111,16 @@ public void run() {
...
-            String operation = item.type;
-            String podName =
-                (item.object.getMetadata() != null)
-                    ? item.object.getMetadata().getName()
-                    : org.apache.commons.lang3.StringUtils.EMPTY;
+            final String operation = item.type;
+            final String podName =
+                Optional.ofNullable(item.object.getMetadata())
+                    .map(V1ObjectMeta::getName)
+                    .orElse("");
```
Comment (ddietterich) [inline src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: What's wrong with StringUtils.empty?

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java [hunk 4]
```diff
@@ -177,14 +210,10 @@ Exception getException() {
...
-    int count = 0;
-    for (PodState podState : podMap.values()) {
-      if (podState == PodState.RUNNING) {
-        count++;
-      }
-    }
+    final long count = podMap.values().stream().filter(PodState.RUNNING::equals).count();
     logger.info("KubePodListener ActivePodCount: {} of {} pods active", count, podMap.size());
-    return count;
+    return (int) count;
```
Comment (ddietterich) [inline src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: Maybe use Math.toIntExact instead of the cast.

**Review by ddietterich — COMMENTED** [review]
Thanks for the cleanup effort! A few comments and a suggested alternative.

**Comment by ddietterich** [comment]
I don't think it is worth trying to unit test this. Seems to me a lot of effort for little benefit. The whole purpose of the code is to interact with K8s, so I believe in testing it with K8s.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java (hunk 3 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: I'd never seen it before. Is there an issue with `""`? I thought that was standard.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java (hunk 4 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: oh nice

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java (hunk 1 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: That works all right, thanks.

File: src/main/java/bio/terra/common/kubernetes/KubePodListener.java (hunk 3 above)
Comment (ddietterich) [reply src/main/java/bio/terra/common/kubernetes/KubePodListener.java]: It is probably bit bumming, but every "" creates a String object. Using EMPTY shares one.

**Review by ddietterich — APPROVED** [review]
I like it!

**Review by tnadler — APPROVED** [review]
Super helpful in making this file understandable, thank you!

**Pull request merged**